### PR TITLE
Add changelog note for CodeQL v2.10.5 deprecation

### PR DESCRIPTION
**Pull request by henrymercer**
I should follow my own instructions! :)

### Merge / deployment checklist

- [ ] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [ ] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

**Comment by henrymercer** [comment]
@aeisenberg Did you mean to approve?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,9 @@ See the [releases page](https://github.com/github/codeql-action/releases) for th
 
 ## [UNRELEASED]
 
-No user facing changes.
+- Add a deprecation warning for customers using CodeQL version 2.11.5 and earlier. These versions of CodeQL were discontinued on 8 November 2023 alongside GitHub Enterprise Server 3.7, and will be unsupported by the next minor release of the CodeQL Action. [#1993](https://github.com/github/codeql-action/pull/1993)
```
Comment (aeisenberg) [inline CHANGELOG.md:7]: Use North American style dates, I think.

```suggestion
- Add a deprecation warning for customers using CodeQL version 2.11.5 and earlier. These versions of CodeQL were discontinued on November, 8 2023 alongside GitHub Enterprise Server 3.7, and will be unsupported by the next minor release of the CodeQL Action, 2.23. [#1993](https://github.com/github/codeql-action/pull/1993)
```

**Review by aeisenberg — COMMENTED** [review]
I did approve, but I'm now seeing two very minor issues.

File: CHANGELOG.md (hunk 1 above)
Comment (aeisenberg) [reply CHANGELOG.md:7]: Nevermind, I take that back. We're using British dates elsewhere in this changelog, so lets just keep it here.

But, would still be nice to explicitly use the version of the action where it is deprecated.

```suggestion
- Add a deprecation warning for customers using CodeQL version 2.11.5 and earlier. These versions of CodeQL were discontinued on 8 November 2023 alongside GitHub Enterprise Server 3.7, and will be unsupported by the next minor release of the CodeQL Action, 2.23. [#1993](https://github.com/github/codeql-action/pull/1993)
```

Comment (henrymercer) [reply CHANGELOG.md:7]:
```suggestion
- Add a deprecation warning for customers using CodeQL version 2.11.5 and earlier. These versions of CodeQL were discontinued on 8 November 2023 alongside GitHub Enterprise Server 3.7, and will be unsupported by CodeQL Action v2.23.0 and later. [#1993](https://github.com/github/codeql-action/pull/1993)
```

**Review by aeisenberg — APPROVED** [review]

**Pull request merged**